plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsMaker.java

```diff
@@ -124,7 +133,27 @@ private TableStatistics makeTableStatistics(IcebergTableHandle tableHandle)
             if (summary.getColumnSizes() != null) {
                 Long columnSize = summary.getColumnSizes().get(fieldId);
                 if (columnSize != null) {
-                    columnBuilder.setDataSize(Estimate.of(columnSize));
+                    if (columnHandle.getBaseType() instanceof FixedWidthType) {
+                        // columnSize is the size on disk and Trino column stats is size in memory.
+                        // The relation between the two is type and data dependent.
+                        // However, Trino currently does not use data size statistics for fixed-width types
+                        // (it's not needed for them), so do not report it at all, to avoid reporting some bogus value.
+                    }
+                    else {
+                        if (idToType.get(columnHandleTuple.getKey()).typeId() == Type.TypeID.FIXED) {
+                            Types.FixedType fixedType = (Types.FixedType) idToType.get(columnHandleTuple.getKey());
+                            columnSize = (long) fixedType.length();
```

Comment:
That should be filled in also when `summary.getColumnSizes().get(fieldId) == null`